account-kit/signer/src/base.ts

```diff
@@ -777,15 +790,21 @@ export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
     bundle,
     orgId,
     idToken,
-  }: Extract<AuthParams, { type: "oauthReturn" }>): Promise<User> =>
-    this.inner.completeAuthWithBundle({
+    isNewUser,
+  }: Extract<AuthParams, { type: "oauthReturn" }>): Promise<User> => {
+    const user = this.inner.completeAuthWithBundle({
       bundle,
       orgId,
       connectedEventName: "connectedOauth",
       authenticatingType: "oauth",
       idToken,
     });
 
+    if (isNewUser) this.store.setState({ isNewUser });
```

Comment:
nit: we might wanna wrap this in a method called `emitNewUserEvent` so that it's a bit clearer that's what this does